### PR TITLE
Feature/1977 smtp validation

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/email/email_service.ml [hunk 1]
```diff
@@ -262,6 +238,88 @@ module Smtp = struct
...
+        ; password
+        ; mechanism
+        ; protocol
+        ; _
+        }
+      =
+      config
+    in
+    let test_email =
+      let sender = "todo@email.com" in
```
Comment (mabiede) [inline pool/app/email/email_service.ml:304]: "todo@email.com"

File: pool/app/email/email_service.ml [hunk 2]
```diff
@@ -262,6 +238,88 @@ module Smtp = struct
...
+        ; username
+        ; password
+        ; mechanism
+        ; protocol
+        ; _
+        }
+      =
+      config
+    in
+    let test_email =
```
Comment (mabiede) [inline pool/app/email/email_service.ml:303]:
```suggestion
  let prepare_test_email
    database_label
    { SmtpAuth.Write.server; port; username; password; mechanism; protocol; _ }
    test_email
    =
    let%lwt default_sender_of_pool = default_sender_of_pool database_label in
    let test_email =
```

File: pool/app/email/email_service.ml [hunk 3]
```diff
@@ -262,6 +238,88 @@ module Smtp = struct
     Lwt.return { sender; reply_to; recipients; subject; body; config }
   ;;
 
+  let prepare database_label ?smtp_auth_id email =
+    let open Utils.Lwt_result.Infix in
+    let%lwt config =
```
Comment (mabiede) [inline pool/app/email/email_service.ml:243]:
```suggestion
    llet%lwt { SmtpAuth.Write.server
            ; port
            ; username
            ; password
            ; mechanism
            ; protocol
            ; _
            }
      =
```

File: pool/app/email/email_service.ml [hunk 4]
```diff
@@ -262,6 +238,88 @@ module Smtp = struct
...
+        ; port
+        ; username
+        ; password
+        ; mechanism
+        ; protocol
+        ; _
+        }
+      =
+      config
+    in
```
Comment (mabiede) [inline pool/app/email/email_service.ml:278]:
```suggestion
```

File: pool/web/handler/admin_settings_smtp.ml [hunk 5]
```diff
@@ -198,7 +221,41 @@ let delete_base location req =
...
+;;
+
+let validate_tenant = validate `Tenant
+
+module Access : sig
+  val index : Rock.Middleware.t
+  val create : Rock.Middleware.t
+  val read : Rock.Middleware.t
+  val update : Rock.Middleware.t
+  val delete : Rock.Middleware.t
```
Comment (mabiede) [inline pool/web/handler/admin_settings_smtp.ml:256]:
```suggestion
  include module type of Helpers.Access
  
```

File: pool/app/email/email.mli [hunk 6]
```diff
@@ -199,6 +199,11 @@ module SmtpAuth : sig
 
   val find_by_label : Pool_database.Label.t -> Label.t -> t option Lwt.t
 
+  val find_full
+    :  Pool_database.Label.t
+    -> Id.t
+    -> (Write.t, Pool_common.Message.error) Lwt_result.t
+
   (* TODO: Can probably be removed *)
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/email/email.mli:207]: removed

**Review by mabiede — APPROVED** [review]

**Pull request merged**